docs/source/migration.mdx

```diff
@@ -407,130 +401,143 @@ interface ContextValue {
   },
+  //highlight-start
   dataSources: (): ContextValue['dataSources'] => {
     return {
       moviesAPI: new MoviesAPI(),
     };
   },
+  //highlight-end
 });
 
 await server.listen();
 ```
 
 </MultiCodeBlock>
 
-<!-- TODO(AS4): Update the next line before release -->
-
-Below is how you write the same code in Apollo Server 4. Note that the `@apollo/datasource-rest` package is now renamed `@apollo/datasource-rest`.
-
-> ⚠️ This feature is in active development, and the below code snippets **do not** currently work as described. [See above for more details](#datasources).
+Below is how you write the same code in Apollo Server 4.
 
 <MultiCodeBlock>
 
-```ts {5, 25-27, 38, 41-43}  title="Apollo Server 4"
-import { RESTDataSource, RESTDataSourceOptions } from '@apollo/datasource-rest';
+```ts title="Apollo Server 4"
+import { RESTDataSource, WillSendRequestOptions } from '@apollo/datasource-rest';
+// KeyValueCache is the type of Apollo server's default cache
+import type { KeyValueCache } from '@apollo/utils.keyvaluecache';
 import { ApolloServer } from '@apollo/server';
 import { startStandaloneServer } from '@apollo/server/standalone';
 
-class MoviesAPI extends RESTDataSource {
-  baseURL = 'https://movies-api.example.com/';
+class MoviesAPI extends RESTDataSource { // highlight-line
+  override baseURL = 'https://movies-api.example.com/';
   private token: string;
 
-  constructor(options: { token: string } & RESTDataSourceOptions) {
-    super(options);  // this should send `cache` through
+  constructor(options: { token: string; cache: KeyValueCache }) {
+    super(options); // this sends our server's `cache` through
     this.token = options.token;
   }
 
-  willSendRequest(request: RequestOptions) {
-    request.headers.set('Authorization', this.token);
+  override willSendRequest(request: WillSendRequestOptions) {
+    request.headers['authorization'] = this.token;
   }
 
-  async getMovie(id: string): Movie {
-    return this.get(`movies/${encodeURIComponent(id)}`);
+  async getMovie(id: string): Promise<Movie> {
+    return this.get<Movie>(`movies/${encodeURIComponent(id)}`);
   }
 }
 
+// highlight-start
 interface ContextValue {
   token: string;
   dataSources: {
     moviesAPI: MoviesAPI;
-  }
-};
+  };
+}
+// highlight-end
 
 const server = new ApolloServer<ContextValue>({
   typeDefs,
   resolvers,
 });
 
-await startStandaloneServer(server, {
+const { url } = await startStandaloneServer(server, {
   context: async ({ req }) => {
     const token = getTokenFromRequest(req);
     const { cache } = server;
     return {
       token,
+      //highlight-start
       dataSources: {
         moviesAPI: new MoviesAPI({ cache, token }),
       },
+      //highlight-end
     };
   },
 });
+
+console.log(`🚀  Server ready at ${url}`);
 ```
 
 </MultiCodeBlock>
 
+> For Apollo Server 4, `apollo-datasource-rest` package has been updated and renamed [`@apollo/datasource-rest`](https://www.npmjs.com/package/@apollo/datasource-rest).
```

Comment:
I feel like this works better before the code block.

Reply:
Oh, I like that; actually, I'll move this all the way up and make it the first thing in the article.